docs/ref/contrib/postgres/search.txt

```diff
@@ -0,0 +1,151 @@
+
+    >>> Entry.objects.annotate(
+    ...     search=SearchVector('body_text', 'blog__tagline'),
+    ... ).filter(search='Cheese')
+    [<Entry: Cheese on Toast recipes>, <Entry: Pizza Recipes>]
+
+The arguments passed can be any :class:`~django.db.models.Expression`, or the
+name of a field. Multiple arguments will be concatenated together using a
+space, so the search document includes both.
+
+``SearchVector`` objects can be combined together, allowing you to reuse
+certain search vectors. For example::
+
+    >>> Entry.objects.annotate(
+    ...     search=SearchVector('body_text') + SearchVector('blog__tagline'),
+    ... ).filter(search='Cheese')
+    [<Entry: Cheese on Toast recipes>, <Entry: Pizza Recipes>]
+
+``SearchQuery``
+===============
+
+.. class:: SearchQuery(config=None)
+
+``SearchQuery`` translates the terms the user provides into a search query
+object it can compare to a search vector. By default, all of the words the user
+provides are passed through the stemming algorithms, and then it looks for
+matches for all of the resulting terms.
+
+If you want to provide more flexibility to the user, for example allowing them
+to find records where ``potato`` OR ``penguin`` are contained in the result, or
+results not containing ``sausage``::
+
+    >>> SearchQuery('potato') & SearchQuery('ireland')  # potato AND ireland
+    >>> SearchQuery('potato') | SearchQuery('penguin')  # potato OR penguin
+    >>> ~SearchQuery('sausage')  # NOT sausage
+
+``SearchRank``
+==============
+
+.. class:: SearchRank(vector, query, weights=None)
+
+So far, we have just returned the results for which any match between the
+vector and the query are possible. It is likely you may wish to order the
+results by some level of relevancy. PostgreSQL provides a ranking function
+which takes into account how often the query terms appear in the document, how
+close together the terms are in the document, and how important is the part of
+the document where they occur. The better the match, the higher the value of
+the rank. We can now order by relevancy::
+
+    >>> vector = SearchVector('body_text')
+    >>> query = SearchQuery('cheese')
+    >>> Entry.objects.annotate(rank=SearchRank(vector, query)).order_by('-rank')
+    [<Entry: Cheese on Toast recipes>, <Entry: Pizza recipes>]
+
+Changing the search configuration
+=================================
+
+You can specify the ``config`` attribute to a ``SearchVector`` and
+``SearchQuery`` to use a different search configuration. This allows you to use
+a different language parsers and dictionaries as defined by the database::
+
+    >>> Entry.objects.annotate(
+    ...     search=SearchVector('body_text', config='french'),
+    ... ).filter(search=SearchQuery('œuf', config='french'))
+    [<Entry: Pain perdu>]
+
+The value of config could also be stored in another column::
+
+    >>> Entry.objects.annotate(
+    ...     search=SearchVector('body_text', config=F('blog__language')),
```

Comment:
I'm getting "can't adapt type 'F'" for this example (doesn't seem to be tested).